lib/src/revset.rs

```diff
@@ -853,6 +855,17 @@ static BUILTIN_FUNCTION_MAP: Lazy<HashMap<&'static str, RevsetFunction>> = Lazy:
             visible_heads: None,
         }))
     });
+    map.insert("coalesce", |diagnostics, function, context| {
+        let ([], args) = function.expect_some_arguments()?;
+        let mut revset = RevsetExpression::none();
+        for arg in args.iter().rev() {
+            revset = Rc::new(RevsetExpression::Coalesce(
+                lower_expression(diagnostics, arg, context)?,
+                revset,
+            ));
```

Comment:
nit: can you add a few `RevsetExpression` construction tests (similar to e.g. `union_all()`)? It might be easier if this function is extracted as `RevsetExpression` method.

btw, this function will leave a redundant `none()` at the bottom expression. I think that's okay, but if we want to simplify the constructed expression, we can either special case `[]` or do recursion in the same way as `union_all()`.